Assets/Scripts/ConnectionManagement/ConnectionState/ClientConnectedState.cs

```diff
@@ -24,20 +24,19 @@ void InjectDependencies(IPublisher<ConnectStatus> connectStatusPublisher)
         public override void OnClientDisconnect(ulong _)
         {
             m_ConnectStatusPublisher.Publish(ConnectStatus.Reconnecting);
-            m_ConnectionManager.ChangeState(ClientReconnecting);
+            StateChangeRequest?.Invoke(ClientReconnecting);
```

Comment:
why conditional operator? I'd assume there IS a callback here. If we assume it's there and it's null, I'd fail loudly (and let it throw an exception)